modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -163,10 +164,10 @@ def grab_nas
       server = Regexp.last_match[1].unpack('C*').join('.')
       port = Regexp.last_match[2].unpack('S')
     end
-    if /[\x00]{16,}(?<ftpuser>[[:print:]]+)[\x00]{16,}(?<ftppass>[[:print:]]+)/ =~ data
+    if data =~ /[\x00]{16,}(?<ftpuser>[[:print:]]+)[\x00]{16,}(?<ftppass>[[:print:]]+)/
```

Comment:
I don't think this will work.  AFAIK, named capturing regular expressions only work and save the named value if the values are on the left hand side of the `=~`.